src/common/transformations/src/transformations/common_optimizations/nop_elimination.cpp

```diff
@@ -443,16 +443,54 @@ pass::EliminateConvert::EliminateConvert() {
         if (!convert) {
             return false;
         }
+
         if (convert->get_input_element_type(0) == convert->get_element_type()) {
             return replace_output_update_name(convert->output(0), convert->input_value(0));
         }
+
+        // Only for some test.
+        if (convert->get_rt_info().count("DisableEliminateUselessConvert") &&
+            convert->get_rt_info()["DisableEliminateUselessConvert"].as<std::string>() == std::string("1")) {
+            return false;
+        }
+
+        // Eliminate useless convert pattern, for example:
+        // Convert[fp16->fp32] -> Convert[fp32->f16]
+        if (convert && convert->get_output_size() > 0u && convert->get_input_size() > 0u) {
+            auto convert2 = ov::as_type_ptr<ov::op::v0::Convert>(
+                convert->get_output_target_inputs(0).begin()->get_node()->shared_from_this());
+            if (convert2 && convert->get_input_element_type(0) == convert2->get_output_element_type(0) &&
```

Comment:
In general such optimization is not safe. Cases like Convert[fp32->fp16] -> Convert[fp16->f32] still needs to be executed explicitly as long as its elemintation will affect model accuracy.

However we still can adjust the pass to the patterns which are safe to remove (then src/dst precisions can be converted to intermediate precision w/o any loss).
@itikhono Do you think it is reasonable extension for the existing EliminateConvert pass?

Reply:
Thanks @dmitry-gorokhov for the clarification.